daemon/prune.go

```diff
@@ -135,24 +135,29 @@ func (daemon *Daemon) ImagesPrune(pruneFilters filters.Args) (*types.ImagesPrune
 		deletedImages := []types.ImageDelete{}
 		refs := daemon.referenceStore.References(dgst)
 		if len(refs) > 0 {
-			if danglingOnly {
-				// Not a dangling image
-				continue
+			shouldDelete := !danglingOnly
+			if !shouldDelete {
+				hasTag := false
```

Comment:
`shouldDelete` needs to stay `false` in that case. I only want to set it to `true` if there's no `NamedTagged`. If I were to set to `true` here it would delete all images.